banish/models.py

```diff
@@ -66,14 +69,52 @@ def is_current(self):
 
     class Meta:
         permissions = (("can_ban_user", "Can Ban User"),)
-        verbose_name = "Banishments"
+        verbose_name = "Banishment"
         verbose_name_plural = "Banishments"
         db_table = 'banishments'
 
 
+class Whitelist(models.Model):
+    id = models.AutoField(primary_key=True)
+
+    whitelist_reason = models.CharField(max_length=255, help_text="Reason for the whitelist?")
+
+    WHITELIST_TYPES = (
+        ('ip-address-whitelist', 'Whitelist IP Address'),
+    )
+
+    type = models.CharField(
+        max_length=20,
+        choices=WHITELIST_TYPES,
+        default=0,
+        help_text='Enter an IP address to whitelist'
+    )
+
+    condition = models.CharField(
+        max_length=255,
+        help_text='Enter an IP to whitelist'
+    )
+
+    def __unicode__(self):
+        return "Whitelisted %s %s " % (self.type, self.condition)
+
+    def __str__(self):
+        return self.__unicode__()
+
+    class Meta:
+        permissions = (("can_whitelist_user", "Can Whitelist User"),)
+        verbose_name = "Whitelist"
+        verbose_name_plural = "Whitelists"
+        db_table = 'whitelists'
+
+
 def _update_cache(sender, instance, **kwargs):
      if instance.type == 'ip-address':
         cache_key = BANISH_PREFIX + instance.condition
-        cache.set(cache_key, "1")  
+        cache.set(cache_key, "1")
+     elif instance.type == 'ip-address-whitelist':
+        cache_key = WHITELIST_PREFIX + instance.condition
+        cache.set(cache_key, "1")        
+
 
 post_save.connect(_update_cache, sender=Banishment)
```

Comment:
@jamonation we also need to connect Whitelist model here, because it will not fire this event when Whitelist will be saved. Ideally, it would be great to have separate function for every Model.